test/libsolidity/SemanticTest.cpp

```diff
@@ -274,6 +310,32 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 	return TestResult::Success;
 }
 
+bool SemanticTest::checkGasCostExpectation(TestFunctionCall& io_test, bool _compileViaYul) const
+{
+	string setting =
+		(_compileViaYul ? "ir"s : "legacy"s) +
+		(m_optimiserSettings == OptimiserSettings::full() ? "Optimized" : "");
+
+	// We don't check gas if enforce gas cost is not active
+	// or in case we run abiencoderv1
+	// or gas used less than threshold for enforcing feature
+	// or setting is "ir" and it's not included in expectations
+	if (
+		!m_enforceGasCost ||
+		solidity::test::CommonOptions::get().useABIEncoderV1 ||
```

Comment:
`CommonOptions::get().useABIEncoderV1` depends only on the command-line option (and not on the pragma in the file), right?

Then I think you should add that case to `CommonOptions::validate()` and just put an assertion here.